### PR TITLE
Add codemod to remove create-reducer

### PR DESCRIPTION
**Pull request by sirreal**
This codemod replaces `createReducer` reducers with simple function reducers with a `switch` statement.

### CODE REVIEW

File: transforms/remove-create-reducer.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+export default function transformer( file, api ) {
```
Comment (sirreal) [inline transforms/remove-create-reducer.js]: This should account for persistence. See https://github.com/Automattic/wp-calypso/blob/898d753d472349bf611509ce128068b38937040e/client/state/utils/create-reducer.js#L29

**Comment by sirreal** [comment]
@sgomes I know you've been improving this. I invite you to take over the branch if you like 👍

**Comment by sgomes** [comment]
> @sgomes I know you've been improving this. I invite you to take over the branch if you like 👍

Consider it taken over! 😄 

The codemod should now handle a larger array of cases and output simpler code. It will require manual review, as it makes a few assumptions (e.g., that `SERIALIZE === 'SERIALIZE'`) and because it produces incomplete output in a few cases (with a `TODO` that we can search for).

I'm going to run a few tests on top of Calypso before declaring this complete.

File: transforms/remove-create-reducer.js (hunk 1 above)
Comment (sgomes) [reply transforms/remove-create-reducer.js]: Done.

**Comment by sgomes** [comment]
I made a bunch more changes to this:
- Handle deconstructs
- Preserve comments (they were removed when the import line was deleted)
- Improve tests
- Transform `createReducerWithValidation` as well
- Apply `withoutPersistence` where needed. We were missing this, and it's pretty important 😕 
- General tweaks and bugfixes

@sirreal I think this is ready for review. The code it generates is sane, and doesn't break any unit tests when applied to Calypso. Should there be a review, or do we just merge and create the PR for Calypso?

File: config.js [hunk 2]
```diff
@@ -1,12 +1,7 @@
-const jscodeshiftArgs = [
-	'--extensions=js,jsx',
-];
+const jscodeshiftArgs = [ '--extensions=js,jsx' ];
```
Comment (sirreal) [inline config.js]: Do we need to handle `ts,tsx` as well now?

Comment (sgomes) [reply config.js]: Possibly, although I'm not sure if the tools support that, nor whether there would be any code changes needed.

In any case, there don't appear to be any reducers in `.ts` files at the moment.

**Comment by sgomes** [comment]
> I can't approve my own PR, but this seems great, especially looking at results in [Automattic/wp-calypso#36470](https://github.com/Automattic/wp-calypso/pull/36470)
> 
> Thanks for pushing it forward @sgomes!
> 
> I think this is ready to merge if you'd like to go ahead 👍

Thanks, @sirreal ! I'll go ahead and merge, then 🙂

**Pull request merged**